### PR TITLE
Debug:xbox_one

### PR DESCRIPTION
**Pull request by tao8687**

### CODE REVIEW

File: modules/canbus/tools/BUILD [hunk 1]
```diff
@@ -61,4 +61,21 @@ cc_binary(
...
+    deps = [
+        "//cyber",
+        "//modules/canbus/vehicle:vehicle_controller_base",
+        "//modules/common/adapters:adapter_gflags",
+        "//modules/control/proto:control_cmd_cc_proto",
+        "//modules/canbus/common:canbus_common",
+        "//modules/canbus/proto:chassis_cc_proto",
+        "//modules/common/configs:vehicle_config_helper",
+        "//modules/common/util:message_util",
+	"@com_github_gflags_gflags//:gflags",
```
Comment (mickeyouyou) [inline modules/canbus/tools/BUILD:77]: codes align

File: modules/canbus/tools/xbox_one.cc [hunk 2]
```diff
@@ -0,0 +1,248 @@
...
+#include <string.h>
+#include <sys/ioctl.h>
+#include <sys/time.h>
+#include <sys/types.h>
+#include <termios.h>
+#include <unistd.h>
+
+#include "cyber/cyber.h"
+#include "modules/canbus/proto/chassis.pb.h"
+#include "modules/canbus/vehicle/vehicle_controller.h"
```
Comment (mickeyouyou) [inline modules/canbus/tools/xbox_one.cc:21]: remove unused include and object.

File: modules/canbus/tools/xbox_one.cc [hunk 3]
```diff
@@ -0,0 +1,248 @@
...
+#include "cyber/common/log.h"
+#include "modules/common/adapters/adapter_gflags.h"
+#include "modules/common/configs/vehicle_config_helper.h"
+#include "modules/common/time/time.h"
+#include "modules/common/util/message_util.h"
+
+namespace {
+// WHILE LOOPce{
+using apollo::canbus::Chassis;
+using apollo::common::VehicleSignal;
```
Comment (mickeyouyou) [inline modules/canbus/tools/xbox_one.cc:33]: remove unused include and object.

File: modules/canbus/tools/xbox_one.cc [hunk 4]
```diff
@@ -0,0 +1,248 @@
...
+    control_command_.set_steering_rate(0.0);
+    control_command_.set_front_wheel_target(0.0);
+    control_command_.set_rear_wheel_target(0.0);
+    control_command_.set_parking_brake(false);
+    control_command_.set_speed(0.0);
+    control_command_.set_acceleration(0.0);
+    control_command_.set_engine_on_off(false);
+    // control_command_.set_driving_mode(Chassis::COMPLETE_MANUAL);
+    control_command_.set_gear_location(Chassis::GEAR_INVALID);
+    control_command_.mutable_signal()->set_turn_signal(
```
Comment (mickeyouyou) [inline modules/canbus/tools/xbox_one.cc:192]: diamond don't have turn signal message, so we can remove it.

File: modules/canbus/tools/xbox_one.cc [hunk 5]
```diff
@@ -0,0 +1,248 @@
...
+  void Stop() {
+    if (is_running_) {
+      is_running_ = false;
+    }
+  }
+  bool IsRunning() const { return is_running_; }
+
+ private:
+  std::unique_ptr<std::thread> xbox_one_thread_;
+  std::shared_ptr<Reader<Chassis>> chassis_reader_;
```
Comment (mickeyouyou) [inline modules/canbus/tools/xbox_one.cc:228]: shared_ptr is point, so it's better to give default value like `nullptr`.

**Review by mickeyouyou — CHANGES_REQUESTED** [review]

File: modules/canbus/tools/xbox_one.cc (hunk 5 above)
Comment (tao8687) [reply modules/canbus/tools/xbox_one.cc:228]: std::shared_ptr<Reader<Chassis>> chassis_reader_=nullptr;
std::shared_ptr<Writer<ControlCommand>> control_command_writer_=nullptr;

Comment (mickeyouyou) [reply modules/canbus/tools/xbox_one.cc]: OK

**Comment by mickeyouyou** [comment]
we also need an script to manager this executable build file.

**Review by tao8687 — COMMENTED** [review]
update

**Review by mickeyouyou — APPROVED** [review]

**Pull request merged**